### PR TITLE
add getRepository function

### PR DESCRIPTION
**Pull request by sladyn98**
- **Please check if the PR fulfills these requirements**

* [X] The commit message follows our guidelines
* [] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
It adds the getRepositories function

- **What is the current behavior?** (You can also link to an open issue here)
Currently we do not fetch any repository data

- **What is the new behavior (if this is a feature change)?**
Now we can possibly add new functionality with repo data

- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

- **Other information**:

### CODE REVIEW

File: app/github.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  for await (const response of octokit.paginate.iterator(
+    'GET /{username}/repos',
+    parameters
+  )) {
+    // do whatever you want with each response
+    console.log(response.name);
+    console.log(response.html_url);
+    console.log(response.forks);
+    console.log(response.stargazers_count);
```
Comment (sladyn98) [inline app/github.js:26]: @und3fined-v01d Do you want this returned like a json object, that you can parse ?

File: app/github.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable no-unused-vars */
+import { Octokit } from '@octokit/core';
```
Comment (und3fined-v01d) [inline app/github.js:3]: use require here

File: app/github.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+const {
+  paginateRest,
+  composePaginateRest,
+} = require('@octokit/plugin-paginate-rest');
+
+async function getRepositories(accessToken) {
+  const octokit = new Octokit({ auth: `${accessToken}` });
```
Comment (und3fined-v01d) [inline app/github.js:11]: we dont need to use the backticks in this line do we?

File: app/github.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  const username = JSON.parse(await octokit.request('GET /user')).login;
+
+  const parameters = {
+    username: `${username}`,
+  };
+
+  for await (const response of octokit.paginate.iterator(
```
Comment (und3fined-v01d) [inline app/github.js:18]: Let us use the `then` and `catch`  as illustrated [here ](https://octokit.github.io/rest.js/v18#pagination)

File: app/github.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    'GET /{username}/repos',
+    parameters
+  )) {
+    // do whatever you want with each response
+    console.log(response.name);
+    console.log(response.html_url);
+    console.log(response.forks);
```
Comment (und3fined-v01d) [inline app/github.js:25]: We can use filter and return these as params. Also use the `camelcase-keys` npm package here or eslint might complain

File: app/github.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import { Octokit } from '@octokit/core';
+
+const {
+  paginateRest,
+  composePaginateRest,
+} = require('@octokit/plugin-paginate-rest');
+
+async function getRepositories(accessToken) {
+  const octokit = new Octokit({ auth: `${accessToken}` });
+  const username = JSON.parse(await octokit.request('GET /user')).login;
```
[the remark this answers is not in the document]
Comment (sladyn98) [reply app/github.js:12]: Oh yeah that was a mistake :P

File: app/github.js (hunk 3 above)
Comment (sladyn98) [reply app/github.js:11]: Yeah we dont

**Pull request closed without merging**